native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageExecutableLocator.java

```diff
@@ -91,29 +91,32 @@ public static File findNativeImageExecutable(Property<JavaLauncher> javaLauncher
             executablePath = metadata.getInstallationPath().file("bin/" + NATIVE_IMAGE_EXE).getAsFile();
         }
 
-        try {
-            if (!executablePath.exists()) {
-                logger.log("Native Image executable wasn't found. We will now try to download it. ");
-                File graalVmHomeGuess = executablePath.getParentFile();
-
-                File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
-                if (!guPath.exists()) {
-                    throw new GradleException("'" + GU_EXE + "' at '" + guPath + "' tool wasn't found. This probably means that JDK at isn't a GraalVM distribution.");
-                }
+        File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
+        if (guPath.exists() && !executablePath.exists()) {
+            logger.log("Native Image executable wasn't found. We will now try to download it. ");
+            File graalVmHomeGuess = executablePath.getParentFile();
+
+            File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
+            if (guPath.exists()) {
                 ExecResult res = execOperations.exec(spec -> {
                     spec.args("install", "native-image");
                     spec.setExecutable(Paths.get(graalVmHomeGuess.getAbsolutePath(), GU_EXE));
                 });
                 if (res.getExitValue() != 0) {
-                    throw new GradleException("Native Image executable wasn't found, and '" + GU_EXE + "' tool failed to install it.");
+                    throw new GradleException("Native Image executable wasn't found, and '" + GU_EXE + "' tool failed to install it.\n" +
+                            "Make sure to declare the GRAALVM_HOME environment variable or install GraalVM with " +
+                            "native-image in a standard location recognized by Gradle Java toolchain support");
                 }
                 diagnostics.withGuInstall();
             }
-        } catch (GradleException e) {
-            throw new GradleException("Determining GraalVM installation failed with message: " + e.getMessage() + "\n\n"
-                                      + "Make sure to declare the GRAALVM_HOME environment variable or install GraalVM with " +
-                                      "native-image in a standard location recognized by Gradle Java toolchain support");
         }
+
+        if (!executablePath.exists()) {
+            throw new GradleException(executablePath + " wasn't found. This probably means that JDK isn't a GraalVM distribution.\n" +
+                    "Make sure to declare the GRAALVM_HOME environment variable or install GraalVM with " +
```

Comment:
```suggestion
                    "Make sure to declare the GRAALVM_HOME or JAVA_HOME environment variables or install GraalVM with " +
```

Reply:
I added the `or JAVA_HOME`, I don't know if it's respected by the Gradle plugin though.
I kept `variable` singular since the user should set either of them but no need to set both.